app/metrics.yaml

```diff
@@ -1562,6 +1562,21 @@ collections:
     notification_emails:
       - fenix-core@mozilla.com
     expires: "2020-09-01"
+  total_collections:
+    type: quantity
+    unit: unit
+    gecko_datapoint: NA
+    send_in_pings:
+      - events
```

Comment:
@mcomella after talking to @sraturi , he made me notice that this would be sent in the `events` ping. Why is that?

We also discussed a bit about using the `counter` metric type for this. I believe that should work, even with the restriction related to not being able to set "0": you would only set the value if there were collections. If no collections were available, you would not set the counter. This means that, for versions of Fenix that have to have the metric there, you can assume the "metric being missing" = 0.

Reply:
> this would be sent in the events ping. Why is that?

@Dexterp37 It's a general use metric so I wanted to include it in the most general ping. I thought that ping was events. My understanding is that Metrics is used for data that doesn't change much (e.g. OS version) and nothing else seemed general use.

> you can assume the "metric being missing" = 0.

If this allows us to determine the number of users with collection 0 then that seems okay to me. I'm concerned that:
- this doesn't seem semantically meaningful
- it forces logic onto the server and the analysis phase while other glean implementations seem to have logic on the client (e.g. timing distribution). I guess the major downside is we can't analyze this with GLAM (when it's available)?

I'm not up-to-date on your discuss with sraturi so I'm not sure if the ideal solution, long term is to create a new metric type. If not, I'm concerned that we won't be the only folks who run into this sort of problem (want to send a non-counting integer where 0 is meaningful) so I feel like we should create a new metric type but, disclaimer, I don't have much context on the existing discussions. :)

Reply:
> > this would be sent in the events ping. Why is that?
> 
> @Dexterp37 It's a general use metric so I wanted to include it in the most general ping.

That's really the [metrics ping](https://mozilla.github.io/glean/book/user/pings/metrics.html).

> I thought that ping was events. My understanding is that Metrics is used for data that doesn't change much (e.g. OS version) and nothing else seemed general use.

Oh, that's interesting. Was it something wrong in our docs that led to this misunderstanding? The metrics ping is meant for all kinds of data, regardless of their frequency of change. In fact, if you don't specify which pings to send stuff on, all the metrics (except the events) go in that ping!

> > you can assume the "metric being missing" = 0.
> 
> If this allows us to determine the number of users with collection 0 then that seems okay to me. I'm concerned that:
> 
>     * this doesn't seem semantically meaningful

@sraturi filed a bug for using `quantity` outside of Geckoview. In addition to that, @grigoryk filed [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1626372) for a similar issue.

This will be looked into, as the process for adding/changing metric type moves on.
Unfortunately, since we only recently introduced the process, we're still going through the backlog, but we're planning on getting to this issue in the upcoming weeks.

>     * it forces logic onto the server and the analysis phase while other glean implementations seem to have logic on the client (e.g. timing distribution). I guess the major downside is we can't analyze this with GLAM (when it's available)?

This is a good point. This is definitely one of the aspects that will be cleared when going through the new metric type process. Depending on how soon you want the data, you may still stick to counter. 

> I'm not up-to-date on your discuss with sraturi so I'm not sure if the ideal solution, long term is to create a new metric type. If not, I'm concerned that we won't be the only folks who run into this sort of problem (want to send a non-counting integer where 0 is meaningful) so I feel like we should create a new metric type but, disclaimer, I don't have much context on the existing discussions. :)

Sure, filing a bug was a good call! And making sure folks don't stumble into the same problems is the reason why we're building Glean in the first place :)

Reply:
> Oh, that's interesting. Was it something wrong in our docs that led to this misunderstanding?

I showed up to a presentation late that talked about the differences in the pings and that's the impression that I got but I will admit that I wasn't paying full attention. 😞